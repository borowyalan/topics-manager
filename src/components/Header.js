import React, { Component } from 'react';
import styled from 'styled-components';

import Text from './Text';
import Section from './Section';

class Header extends Component {
    render() {
        return (
            <Section>
                <Centered>
                    <Text type="header" margin="30px">
                        Stwórz Workshop!
                    </Text>
                    <Text type="secondary" margin="0 50px 40px">
                        Scratch the furniture i like big cats and i can not lie.
                        Making sure that fluff gets into the owner’s eyes ask
                        for petting then cats take over the world and need to
                        chase tail, jump launch to pounce upon little yarn
                        mouse, bare fangs at toy run hide in litter box until
                        treats are fed. Flex claws on the humans belly and purr
                        like a lawnmower meow to be let out, drink water out of
                        the faucet. I’m going to lap some water out of my
                        master’s cup meow lick yarn hanging out of own butt so
                        cough yet i’m going to lap some water out of my master’s
                        cup meow all of a sudden cat goes crazy.
                    </Text>
                </Centered>
            </Section>
        );
    }
}

const Centered = styled.div`
    text-align: center;
`;

export default Header;
