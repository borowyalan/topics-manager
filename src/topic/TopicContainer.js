import React from 'react';
import PropTypes from 'prop-types';
import TopicsList from './components/TopicsList';
import { getTopics } from '../actions/topic.actions';
import { connect } from 'react-redux';
import TopicModel from '../shared/models/TopicModel';
import Container from '../shared/components/Container';


// tests for actions & thunk's dispatching
class TopicContainer extends React.Component {

    componentDidMount() {
        this.props.getTopics();
    }

    render() {
        const { topics, pending, error } = this.props;
        return (
            <Container type="section">
                {error && <p>{error}</p>}
                {!error && <TopicsList pending={pending} topics={topics}/>}
            </Container>
        );
    }
}

const mapStateToProps = ({ topic }) => ({
    topics: topic.topics,
    pending: topic.pending,
    error: topic.error
});

const mapDispatchToProps = dispatch => ({
    getTopics: () => dispatch(getTopics())
});

TopicContainer.propTypes = {
    getTopics: PropTypes.func.isRequired,
    topics: PropTypes.arrayOf(PropTypes.instanceOf(TopicModel)),
    pending: PropTypes.bool.isRequired,
    error: PropTypes.any
};

export default connect(mapStateToProps, mapDispatchToProps)(TopicContainer);
