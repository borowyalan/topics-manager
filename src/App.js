import React, { Component } from 'react';
import { injectGlobal } from 'styled-components';
import TopBar from './components/TopBar';
import WorkshopForm from './components/WorkshopForm';
import Header from './components/Header';
import TopicContainer from './topic/TopicContainer';

class App extends Component {
    render() {
        return (
            <React.Fragment>
                <TopBar />
                <Header />
                <main>
                    <WorkshopForm />
                    <TopicContainer />
                </main>
            </React.Fragment>
        );
    }
}

/* eslint no-unused-expressions: 0 */
injectGlobal`
    @import url('https://fonts.googleapis.com/css?family=Lato');

    html {
        box-sizing: border-box;
    }

    *,
    *:after,
    *:before {
        box-sizing: inherit;
    }

    body {
        font-family: 'Lato', sans-serif;
    }
`;

export default App;
