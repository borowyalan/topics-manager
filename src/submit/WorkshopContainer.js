import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { submitTopic } from '../actions/topic.actions';
import WorkshopForm from '../components/WorkshopForm';
import { GithubUser } from '../shared/models/github-user';

class WorkshopContainer extends React.Component {
    render() {
        return (
            <div>
                <WorkshopForm
                    logged={this.props.logged}
                    onClick={this.props.submitTopic}
                    author={this.props.author}
                />
            </div>
        );
    }
}

const mapDispatchToProps = dispatch => ({
    submitTopic: (topic, author) => dispatch(submitTopic(topic, author)),
});

const mapStateToProps = state => ({
    author: state.auth.user,
    logged: state.auth.logged,
});

WorkshopContainer.propTypes = {
    submitTopic: PropTypes.func.submitTopic,
    author: PropTypes.instanceOf(GithubUser).isRequired,
    logged: PropTypes.bool.isRequired,
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(WorkshopContainer);
