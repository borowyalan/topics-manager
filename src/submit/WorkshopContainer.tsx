import React, { SFC } from 'react';
import { connect } from 'react-redux';
import { submitTopic } from '../actions/topic.actions';
import WorkshopForm from '../components/WorkshopForm';
import { TopicMetadata } from '../shared/models/topic-metadata';
import { User } from '../shared/models/user';

export interface WorkshopContainerProps {
    submitTopic: (topic: TopicMetadata, author: User) => undefined,
    author: User,
    logged: boolean,
}

export const WorkshopContainer: SFC<WorkshopContainerProps> = (props) =>
    <WorkshopForm
        logged={props.logged}
        onClick={props.submitTopic}
        author={props.author}/>;

const mapDispatchToProps = dispatch => ({
    submitTopic: (topic, author) => dispatch(submitTopic(topic, author)),
});

const mapStateToProps = state => ({
    author: state.auth.user,
    logged: state.auth.logged,
});

export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(WorkshopContainer);
