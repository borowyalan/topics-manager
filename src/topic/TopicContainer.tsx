import React, { SFC } from 'react';
import { connect } from 'react-redux';
import Section from '../components/Section';
import { Topic } from '../shared/models/topic';
import TopicsList from './components/TopicsList';

export interface TopicContainerProps {
    topics: Topic[],
    pending: boolean,
    error?: any,
}

export const TopicContainer: SFC<TopicContainerProps> = ({ topics, pending, error }) => (
    <Section>
        {error && <p>{error}</p>}
        {!error && <TopicsList pending={pending} topics={topics}/>}
    </Section>
);

const mapStateToProps = ({ topic }) => ({
    topics: topic.topics,
    pending: topic.pending,
    error: topic.error,
});

export default connect(mapStateToProps)(TopicContainer);
