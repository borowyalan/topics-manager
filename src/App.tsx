import React, { Component } from 'react';
import { injectGlobal } from 'styled-components';
import Header from './components/Header';
import TopBar from './components/TopBar';

import AuthCallbackContainer from './shared/components/AuthCallbackContainer';
import Database from './shared/components/Database';
import WorkshopContainer from './submit/WorkshopContainer';
import TopicContainer from './topic/TopicContainer';

class App extends Component {
    public render() {
        return (
            <React.Fragment>
                <TopBar/>
                <Header/>
                <main>
                    <WorkshopContainer/>
                    <TopicContainer/>
                </main>
                <AuthCallbackContainer/>
                <Database/>
            </React.Fragment>
        );
    }
}

// tslint:disable-next-line
injectGlobal`
    @import url('https://fonts.googleapis.com/css?family=Lato:300,400');
    @import url('https://fonts.googleapis.com/css?family=Fira+Sans:300');
    
    html {
        box-sizing: border-box;
    }

    *,
    *:after,
    *:before {
        box-sizing: inherit;
    }

    body {
        font-family: 'Lato', sans-serif;
    }
`;

export default App;
