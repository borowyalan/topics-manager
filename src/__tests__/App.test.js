import React from 'react';
import App from '../App';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import TopicContainer from '../topic/TopicContainer';
import Header from '../components/Header';
import WorkshopContainer from '../submit/WorkshopContainer';

configure({adapter: new Adapter()});

describe('Should render <App />', () => {
    const wrapper = shallow(<App/>);
    it('should contain sections', () => {
        expect(wrapper.find(Header).exists()).toBe(true);
        expect(wrapper.find(WorkshopContainer).exists()).toBe(true);
        expect(wrapper.find(TopicContainer).exists()).toBe(true);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
