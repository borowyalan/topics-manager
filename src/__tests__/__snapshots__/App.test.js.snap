// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render <App /> should match snapshot 1`] = `
<React.Fragment>
  <Connect(TopBar) />
  <Header />
  <main>
    <WorkshopForm />
    <Connect(TopicContainer) />
  </main>
</React.Fragment>
`;
